src/net/peer/connect/connect.rs

```diff
@@ -270,3 +177,20 @@ fn validate_connect_request<UID: Uid>(
     }
     Ok(())
 }
+
+
+/// Starts p2p rendezvous connect.
+///
+/// Gets peer info from rendezvous relay channel and sends connected tcp stream via `out`
+/// channel.
+pub fn rendezvous_connect(
```

Comment:
This function was a bit confusing for me - the way it returns its result via a `oneshot` rather than just directly, and the way it returns `()` for an error. Then I realized it's because it gets turned into a background task with `handle.spawn`.

Since the only point of the future returned here is to be passed to `handle.spawn` I'd prefer it if either (a) the result being wrapped and sent through the channel happened over where `handle.spawn` gets called or (b) `handle.spawn` got called in this function.

Also the `oneshot` channel should probably carry a `Result`. There's also the `.log_error` method for use by daemon-like futures if you don't want to pass the error back.